### PR TITLE
Refactor update-dependencies 1/2

### PR DESCRIPTION
**Pull request by Roming22**
* Use pipelineRef and taskRef
* Start work to make the task generic enough so it can be submitted to redhat-appstudio/build-definitions

Note that the Task is currently incomplete as it does not push the changes made by the script back to the repository.

### CODE REVIEW

File: .tekton/tasks/update-repository.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    - name: TARGET_BRANCH
+      description: Name of the branch which is modified by the Pipeline
+    - name: TARGET_GH_NAME
+      description: Name of the repository which is modified by the Pipeline
+    - name: TARGET_GH_OWNER
+      description: Owner of the repository which is modified by the Pipeline
+    - name: TARGET_GH_URL
+      description: URL of github repository which is modified by the Pipeline
+    - name: SCRIPT_IMAGE
+      description: Image reference to be instanciation to execute the script
```
Comment (ramessesii2) [inline .tekton/tasks/update-repository.yaml:27]: The description is a bit unclear. I'll suggest something on the lines -
```suggestion
      description: Image reference containing a script that is executed to find updates
```

File: .tekton/tasks/update-repository.yaml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      description: ID of Github app used for updating PR
+      default: "305606"
+    - name: GITHUB_APP_INSTALLATION_ID
+      description: Installation ID of Github app in the organization
+      default: "35269675"
+    - name: GIT_IMAGE
+      description: Image reference containing the git command
+      default: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8:v1.8.2-8@sha256:a538c423e7a11aae6ae582a411fdb090936458075f99af4ce5add038bb6983e8
+    - name: GIT_USER
+      description: Username to appear in the commit.
```
Comment (ramessesii2) [inline .tekton/tasks/update-repository.yaml:46]:
```suggestion
      description: Username to appear in the commit
```

File: .tekton/tasks/update-repository.yaml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      default: "Tekton CI"
+    - name: GIT_EMAIL
+      description: Email to appear in the commit.
```
Comment (ramessesii2) [inline .tekton/tasks/update-repository.yaml:49]:
```suggestion
      description: Email to appear in the commit
```

**Review by ramessesii2 — CHANGES_REQUESTED** [review]
Overall looks good to me. Few minor comments though.

File: .tekton/tasks/update-repository.yaml (hunk 2 above)
Comment (Roming22) [reply .tekton/tasks/update-repository.yaml]: Fixed

File: .tekton/tasks/update-repository.yaml (hunk 3 above)
Comment (Roming22) [reply .tekton/tasks/update-repository.yaml]: Fixed

File: .tekton/tasks/update-repository.yaml [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: update-repository
+spec:
+  description: |
+    Clones a repository, runs script in 'SCRIPT' parameter, and generates a pull-request to the repository if a change is detected.
+  params:
+    - name: COMMIT_BRANCH
+      description: Name of the branch which that holds the update created by the Pipeline
```
[the remark this answers is not in the document]
Comment (Roming22) [reply .tekton/tasks/update-repository.yaml]: Fixed

File: .tekton/tasks/update-repository.yaml (hunk 1 above)
Comment (Roming22) [reply .tekton/tasks/update-repository.yaml]: Description updated. The script does not have to be in the image. The image is just used to run it.

**Review by ramessesii2 — APPROVED** [review]
Thanks Romain for incorporating the requested changes.
I'll wait for @xinredhat 's reveiew before merging it.

File: .tekton/tasks/update-repository.yaml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+          value: $(params.TARGET_GH_URL)
+      script: |
+        WORK_DIR="${PWD}/${TARGET_GH_OWNER}/${TARGET_GH_NAME}"
+        if [ -e "${WORK_DIR}" ]; then
+          echo "Clean checkout of '${TARGET_GH_URL}/${TARGET_BRANCH}' in '${WORK_DIR}'"
+          cd "${WORK_DIR}"
+          git clean -d --force
+          git reset --hard
+          git checkout "${TARGET_BRANCH}"
+        else
```
Comment (xinredhat) [inline .tekton/tasks/update-repository.yaml:77]: This block appears unnecessary. we can create volume by using  [emptyDir](https://github.com/redhat-appstudio/build-definitions/blob/main/task/update-infra-deployments/0.1/update-infra-deployments.yaml#L38-L44)  in task. The reason is that 1. it doesn't need to share data/files between tasks. 2. each task creates isolated volume storage if using `emptyDir`

File: .tekton/pipeline/update-repository.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          value: ./developer/images/dependencies-update/hack/bin/update.sh
+        - name: SCRIPT_ARGS
+          value:
+            - --task
+            - update_binaries
+            - --workspace_dir
+            - "."
+      workspaces:
+        - name: workdir
+          workspace: workdir
```
Comment (xinredhat) [inline .tekton/pipeline/update-repository.yaml:41]: it doesn't need the workspace if we use this [way](https://github.com/redhat-appstudio/build-definitions/blob/main/task/update-infra-deployments/0.1/update-infra-deployments.yaml#L38-L44) to create volume in task

File: .tekton/update-dependencies.yaml [hunk 7]
```diff
@@ -21,219 +24,8 @@ spec:
...
-            value: $(params.source_branch)
-          - name: HEAD
-            value: robot/$(params.source_branch)/update_binaries
-          - name: BODY
-            value: Generated by developer/images/dependencies-update/hack/bin/update.sh
-          - name: TITLE
-            value: Update binaries
   workspaces:
-    - name: source
+    - name: workdir
```
Comment (xinredhat) [inline .tekton/update-dependencies.yaml:28]: same as above, it doesn't need to create workspace

File: .tekton/tasks/update-repository.yaml (hunk 5 above)
Comment (Roming22) [reply .tekton/tasks/update-repository.yaml:77]: The task may (will) be called multiple times in the same pipeline. The shared volume will prevent having to clone the repository each time.

File: .tekton/update-dependencies.yaml (hunk 7 above)
Comment (Roming22) [reply .tekton/update-dependencies.yaml:28]: See previous answer about the task being run multiple times in a single pipeline.

**Pull request merged**